test.js

```diff
@@ -71,3 +71,11 @@ test('type inference', t => {
 	t.is(fn({argv: ['5'], inferType: true}, {string: ['foo']}).input[0], 5);
 	t.is(fn({argv: ['5'], inferType: true}, {string: ['_', 'foo']}).input[0], 5);
 });
+
+test('stdinOrInput', async t => {
+	const stdin = await execa.shell('echo \'foo bar\' | ./fixture.js');
+	const input = await execa('./fixture.js', ['foo', 'bar']);
+
+	t.is(stdin.stdout, 'meow\nfoo bar\n');
+	t.is(input.stdout, 'meow\nfoo\nbar');
```

Comment:
That's not what I mean. Usually programs that accept _filenames_ as arguments are the ones that employ the use of `stdin`.

For instance:

```shell
grep "text" some-file.txt
echo "some text" | grep "text"

sed 's/some/thing/' some-file.txt
echo "some text" | sed 's/some/thing/'
```

Instead, this PR seems to support the usage of:

```shell
cat > some-prog.js <<EOF
// ...
console.log(meow.input);
// ...
EOF

some-prog.js This Is Some Text
# doing the same as
echo This Is Some Text | some-prog.js
```

Which is going to confuse a lot of people.